protected/humhub/components/ModuleManager.php

```diff
@@ -213,23 +216,91 @@ public function register($basePath, $config = null)
         Yii::$app->setModule($config['id'], $moduleConfig);
 
         // Register Event Handlers
-        if (isset($config['events'])) {
-            foreach ($config['events'] as $event) {
-                $eventClass = $event['class'] ?? $event[0];
-                $eventName = $event['event'] ?? $event[1];
-                $eventHandler = $event['callback'] ?? $event[2];
+        $events = $config['events'] ?? null;
+        $strict = $config['strict'] ?? false;
+        while (!empty($events)) {
+            $error = static function (string $message, bool $throw = false) use (&$config, $basePath) {
+                $message = "Configuration at $basePath has an invalid event configuration: $message";
+
+                if ($throw) {
+                    throw new InvalidConfigException($message);
+                }
+
+                Yii::warning($message, $config['id']);
+            };
+
+            if (!ArrayHelper::isTraversable($events)) {
+                $error('events must be traversable', $strict);
+                break;
+            }
+
+            $getProperty = static function ($event, &$var, string $property, int $index, bool $throw = false) use ($error): bool {
+
+                $var = $event[$property] ?? $event[$index] ?? null;
+
+                if (empty($var)) {
+                    $error("required property '$property' missing!", $throw);
+                    return false;
+                }
+
+                return true;
+            };
+
+            foreach ($events as $event) {
+                if (empty($event)) {
```

Comment:
It might be good to have a private method e.g. `validateEventCallback($event)`?

Reply:
rather than extracting `validateEventCallback($event)`, I've extracted the entire `registerEventHandlers()`. As such I could simplify the layout of the code a bit. Do you want to take another look?